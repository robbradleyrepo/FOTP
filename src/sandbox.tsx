/* eslint-disable simple-import-sort/imports */

// Import for side effects (`configureXYZ(...)`)
import "../pages/_app";

import Sandbox from "@sss/apollo/testing";
import * as ecommerce from "@sss/ecommerce/testing";

import PageProviders from "./ui/providers";
import introspectionResultData from "../generated/config/fragment-types.json";
import { I18N } from "./config";

export const anonymous = new Sandbox({
  Wrapper: PageProviders,
  globalMocks: [
    ecommerce.mockedResponses.checkout.none,
    ecommerce.mockedResponses.collection.core,
  ],
  introspectionResultData,
  locale: I18N.fallbackLocale,
});

export default anonymous;
