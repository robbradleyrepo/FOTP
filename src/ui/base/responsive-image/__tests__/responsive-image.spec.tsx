import React from "react";

import IMG from "@/modules/home/assets/HERO.jpg";

import sandbox from "../../../../sandbox";
import ResponsiveImage, { getNextImageUrl } from "../";

test("<Image /> default loader route generation", async () => {
  // Why? We've copy+pasted the default loader logic to out of Next.js for internal use.
  // If this snapshot begins to fail, then the route generation logic has changed or the deviceSizes config.
  const { component } = await sandbox.component({
    Component: (
      <ResponsiveImage
        alt="img"
        sizes={{ width: ["100vw", "50vw", "33vw"] }}
        priority
        {...IMG}
      />
    ),
  });

  const img = await component.findByAltText("img");

  expect(img).toMatchSnapshot();
});

describe("getNextImageUrl generates expected routes", () => {
  it("should generate with default quality of 75", () => {
    expect(getNextImageUrl({ src: IMG.src, width: 100 })).toEqual(
      "/_next/image?url=%2Fexample%2Fimage.jpg&w=100&q=75"
    );
  });

  it("should generate with specified quality", () => {
    expect(getNextImageUrl({ quality: 50, src: IMG.src, width: 100 })).toEqual(
      "/_next/image?url=%2Fexample%2Fimage.jpg&w=100&q=50"
    );
  });
});
