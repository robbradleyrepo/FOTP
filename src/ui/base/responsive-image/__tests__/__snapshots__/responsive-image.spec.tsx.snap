// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> default loader route generation 1`] = `
<img
  alt="img"
  class="responsive-image___StyledStyledComponentsHelper-sc-jffgla-0"
  data-nimg="responsive"
  decoding="async"
  sizes="(min-width:768px) 33vw, (min-width:480px) 50vw, 100vw"
  src="/_next/image?url=%2Fexample%2Fimage.jpg&w=3840&q=75"
  srcset="/_next/image?url=%2Fexample%2Fimage.jpg&w=256&q=75 256w, /_next/image?url=%2Fexample%2Fimage.jpg&w=384&q=75 384w, /_next/image?url=%2Fexample%2Fimage.jpg&w=640&q=75 640w, /_next/image?url=%2Fexample%2Fimage.jpg&w=750&q=75 750w, /_next/image?url=%2Fexample%2Fimage.jpg&w=828&q=75 828w, /_next/image?url=%2Fexample%2Fimage.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fexample%2Fimage.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fexample%2Fimage.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fexample%2Fimage.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fexample%2Fimage.jpg&w=3840&q=75 3840w"
  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
/>
`;
